tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -2,42 +2,73 @@
 //
 // SPDX-License-Identifier: MPL-2.0
 
+#include <power_grid_model/main_core/input.hpp>
 #include <power_grid_model/optimizer/tap_position_optimizer.hpp>
 
 #include <doctest/doctest.h>
 
 namespace pgm_tap = power_grid_model::optimizer::tap_position_optimizer;
+namespace main_core = power_grid_model::main_core;
 
 namespace power_grid_model {
 
+namespace {
+using TestComponentContainer = Container<Line, Link, Node, Transformer, Source>;
+using TestState = main_core::MainModelState<TestComponentContainer>;
+} // namespace
+
 TEST_CASE("Test Transformer ranking") {
-    // ToDo: The grid from OntNote page
+    // The grid from OntNote page
```

Comment:
```suggestion
    // The grid from OneNote page
```
in addition, this is quite a poor explanation to anyone from outside